xtask/src/changelog.rs

```diff
@@ -0,0 +1,465 @@
+        let to_search = &remaining[prefix_len..];
+
+        match to_search.match_indices(self.prefix).next() {
+            None => {
+                self.haystack = None;
+                Some(remaining)
+            }
+            Some((idx, _match)) => {
+                let length = idx + prefix_len;
+                self.current_pos += length;
+                Some(&remaining[..length])
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod test_split_prefix_inclusive {
+    #[collapse_debuginfo(yes)]
+    macro_rules! assert_chunks {
+        ($prefix: expr, $haystack: expr, $expected: expr $(,)?) => {
+            assert_eq!(
+                super::SplitPrefixInclusive::new($prefix, $haystack).collect::<Vec<_>>(),
+                $expected.into_iter().collect::<Vec<_>>(),
+            );
+        };
+    }
+
+    #[test]
+    fn it_works() {
+        assert_chunks! {
+            "\n@@",
+            "
+@@ -1,4 +1,5 @@
+ <!--
++    This change should be accepted.
+     Pad out some changes here so we force multiple hunks in a diff.
+     Pad out some changes here so we force multiple hunks in a diff.
+     Pad out some changes here so we force multiple hunks in a diff.
+@@ -17,6 +18,7 @@
+ - Pad out some changes here so we force multiple hunks in a diff.
+ - Pad out some changes here so we force multiple hunks in a diff.
+ - Pad out some changes here so we force multiple hunks in a diff.
++- This change should be accepted.
+\u{0020}
+ ## Recently released
+\u{0020}
+@@ -26,6 +28,7 @@
+ - Pad out some changes here so we force multiple hunks in a diff.
+ - Pad out some changes here so we force multiple hunks in a diff.
+ - Pad out some changes here so we force multiple hunks in a diff.
++- This change was added after release, reject me!
+\u{0020}
+ ## An older release
+\u{0020}
+",
+        [
+                "
+@@ -1,4 +1,5 @@
+ <!--
++    This change should be accepted.
+     Pad out some changes here so we force multiple hunks in a diff.
+     Pad out some changes here so we force multiple hunks in a diff.
+     Pad out some changes here so we force multiple hunks in a diff.",
+                "
+@@ -17,6 +18,7 @@
+ - Pad out some changes here so we force multiple hunks in a diff.
+ - Pad out some changes here so we force multiple hunks in a diff.
+ - Pad out some changes here so we force multiple hunks in a diff.
++- This change should be accepted.
+\u{0020}
+ ## Recently released
+\u{0020}",
+            "
+@@ -26,6 +28,7 @@
+ - Pad out some changes here so we force multiple hunks in a diff.
+ - Pad out some changes here so we force multiple hunks in a diff.
+ - Pad out some changes here so we force multiple hunks in a diff.
++- This change was added after release, reject me!
+\u{0020}
+ ## An older release
+\u{0020}
+",
+        ]
+                }
+    }
+}
+
+#[cfg(test)]
+mod test_hunks_in_a_released_section {
+    #[collapse_debuginfo(yes)]
```

Comment:
What dafaq is this?

Reply:
It's a [dissting](https://doc.rust-lang.org/reference/attributes/debugger.html#the-collapse_debuginfo-attribute)!

When macros are used outside the crate defining them, this is the default. _Within_ a crate, however, compiler errors spans, by default, will point to expanded code, not use sites. I felt it was nicer to do this from a testing standpoint, but it's not absolutely necessary, as the tests are working.

I just noticed we don't seem to have coverage for `xtask` unit tests in CI… 🤔